src/Components/DataHistoryExplorer/Internal/TimeSeriesTwinCallout/TimeSeriesTwinCallout.types.ts

```diff
@@ -0,0 +1,41 @@
+import {
+    IButtonStyles,
+    ICalloutContentStyles,
+    IStyle,
+    IStyleFunctionOrObject,
+    ITheme,
+    IToggleStyles
+} from '@fluentui/react';
+import { IDataHistoryTimeSeriesTwin } from '../../../../Models/Constants';
+
+export interface ITimeSeriesTwinCalloutProps {
+    timeSeriesTwin?: IDataHistoryTimeSeriesTwin;
+    target: string;
+    onDismiss?: () => void;
+    onPrimaryActionClick: (timeSeriesTwin: IDataHistoryTimeSeriesTwin) => void;
```

Comment:
isn't this always an "add" operation? if so, can we rename to `onAdd`? If not, shouldn't we pass the operation type so the parent knows what to do with the information?

Reply:
it is either "add" or "edit". Since we pass the timeSeriesTwin object as param, the consumer would know what to do with this information via the selected time series twin id if exist. you can check it out [here](https://github.com/microsoft/iot-cardboard-js/pull/807/files#diff-50759147c90dc1b8570a05ea0cae82a017031e59269992398ccd1ea23d0abf0eR57).